src/functions.php

```diff
@@ -13,7 +13,9 @@
 function connect($url, array $subProtocols = [], $headers = [], LoopInterface $loop = null) {
     $loop = $loop ?: ReactFactory::create();
 
-    $connector = new Connector($loop);
+    $connector = new Connector($loop, new \React\Socket\Connector($loop, [
+        'timeout' => 20
```

Comment:
Would it be better to put the default timeout in the `Connector` class?

Reply:
That's a much better idea!  😀